buku

```diff
@@ -3134,84 +3134,11 @@ n: don't add parent folder as tag
             index: Optional[int] = None,
             url: Optional[str] = None,
             shorten: bool = True) -> Optional[str]:
-        """Shorten a URL using Google URL shortener.
+        """Shorten/expand a URL using the tny.im service.
 
-        Parameters
-        ----------
-        index : int, optional (if URL is provided)
-            DB index of the bookmark with the URL to shorten. Default is None.
-        url : str, optional (if index is provided)
-            URL to shorten.
-        shorten : bool
-            True to shorten, False to expand. Default is False.
-
-        Returns
-        -------
-        str
-            Shortened url on success, None on failure.
+        Tny.im is no longer available; don't use this method.
         """
-
-        global MYPROXY
-
-        if not index and not url:
-            LOGERR('Either a valid DB index or URL required')
-            return None
-
-        if index:
-            with self.lock:
-                self.cur.execute('SELECT url FROM bookmarks WHERE id = ? LIMIT 1', (index,))
-                results = self.cur.fetchall()
-            if not results:
-                return None
-
-            url = results[0][0]
-
-        from urllib.parse import quote_plus as qp
-
-        url = url or ''
-        urlbase = 'https://tny.im/yourls-api.php?action='
-        if shorten:
-            _u = urlbase + 'shorturl&format=simple&url=' + qp(url)
-        else:
-            _u = urlbase + 'expand&format=simple&shorturl=' + qp(url)
-
-        if MYPROXY is None:
-            gen_headers()
-
-        ca_certs = os.getenv('BUKU_CA_CERTS', default=CA_CERTS)
-        if MYPROXY:
-            manager = urllib3.ProxyManager(
-                MYPROXY,
-                num_pools=1,
-                headers=MYHEADERS,
-                cert_reqs='CERT_REQUIRED',
-                ca_certs=ca_certs)
-        else:
-            manager = urllib3.PoolManager(num_pools=1,
-                                          headers={'User-Agent': USER_AGENT},
-                                          cert_reqs='CERT_REQUIRED',
-                                          ca_certs=ca_certs)
-
-        try:
-            r = manager.request(
-                'POST',
-                _u,
-                headers={
-                    'content-type': 'application/json',
-                    'User-Agent': USER_AGENT}
-            )
-        except Exception as e:
-            LOGERR(e)
-            manager.clear()
-            return None
-
-        if r.status != 200:
-            LOGERR('[%s] %s', r.status, r.reason)
-            return None
-
-        manager.clear()
-
-        return r.data.decode(errors='replace')
+        warn('\'BukuDb.tnyfy_url()\' no longer works due to the takedown of tny.im service.', DeprecationWarning)
```

Comment:
Replaced functionality implementation with a deprecation warning

(Originally it returned `None` on failure, so the API compatibility was nominally retained.)